src/components/router/router.ts

```diff
@@ -0,0 +1,116 @@
+export class Router {
+    private routes: { [key: string]: Function; } = {};
+    private fallbackRoute: Function | null = null;
+
+    public static getUrlParams(): URLSearchParams {
+        return new URLSearchParams(window.location.search)
+    }
+
+    public static getUrlParam(key: string, defaultValue: string): string {
+        return this.getUrlParams().get(key) || defaultValue;
+    }
+
+    public static setUrlParam(key: string, value: string) {
+        const params = Router.getUrlParams();
+        params.set(key, value)
+        Router.redirectTo('?' + params.toString())
+    }
+
+    public static addUrlParamValue(key: string, value: string) {
+        const params = Router.getUrlParams();
+        const oldValue = params.get(key);
+        if (!oldValue) {
+            return Router.setUrlParam(key, value);
+        }
+        params.set(key, oldValue + ',' + value)
+        Router.redirectTo('?' + params.toString())
+    }
+
+    public static removeUrlParamValue(key: string, value: string) {
+        const params = Router.getUrlParams();
+        const oldValue = params.get(key)?.split(',');
```

Comment:
If you use something twice, use const for it: `const targetParam = params.get(key)`